TeachingRecordSystem/src/TeachingRecordSystem.Core/Jobs/EwcWalesImport/InductionImporter.cs

```diff
@@ -0,0 +1,452 @@
+                Guid itrId = Guid.NewGuid();
+                var itrFailureMessage = new StringBuilder();
+                using var rowTransaction = crmQueryDispatcher.CreateTransactionRequestBuilder();
+
+                try
+                {
+                    var createIntegrationTransactionRecord = new CreateIntegrationTransactionRecordQuery()
+                    {
+                        Id = itrId,
+                        IntegrationTransactionId = IntegrationTransactionId,
+                        Reference = totalRowCount.ToString(),
+                        PersonId = null,
+                        InitialTeacherTrainingId = null,
+                        QualificationId = null,
+                        InductionId = null,
+                        InductionPeriodId = null,
+                        DuplicateStatus = null,
+                        FileName = fileName
+                    };
+                    rowTransaction.AppendQuery(createIntegrationTransactionRecord);
+
+                    var lookupData = await GetLookupDataAsync(row);
+                    var validationFailures = Validate(row, lookupData);
+                    personId = lookupData.Person?.ContactId;
+                    inductionId = lookupData.Induction?.Id;
+
+                    //append non processable errors to list of failures that will be a line in
+                    //the IntegrationTransaction (job) failuremessage field.
+                    if (validationFailures.Errors.Any())
+                    {
+                        failureRowCount++;
+                        foreach (var error in validationFailures.Errors)
+                        {
+                            failureMessage.AppendLine(error);
+                            itrFailureMessage.AppendLine(error);
+                        }
+
+                        foreach (var validationMessage in validationFailures.ValidationFailures)
+                        {
+                            itrFailureMessage.AppendLine(validationMessage);
+                            failureMessage.AppendLine(validationMessage);
+                        }
+                    }
+                    else
+                    {
+                        // if contact does not have an associated induction - create one with the data from the imported file row
+                        // else if there is an associated induction update status & passed date with the data from the imported file row
+                        if (!inductionId.HasValue)
+                        {
+                            inductionId = Guid.NewGuid();
+                            var createInductionQuery = new CreateInductionQuery()
+                            {
+                                Id = inductionId.Value,
+                                PersonId = personId,
+                                StartDate = DateTime.ParseExact(row.StartDate, "dd/MM/yyyy", CultureInfo.InvariantCulture, DateTimeStyles.None),
```

Comment:
We could put `"dd/MM/yyyy"` into a `const` and reference that rather than repeating everywhere.